buildSrc/src/main/kotlin/splunk.publish-conventions.gradle.kts

```diff
@@ -0,0 +1,104 @@
+import org.gradle.api.publish.maven.MavenPublication
+import java.net.URI
+
+plugins {
+    id("com.android.library")
+    id("maven-publish")
+    id("signing")
+}
+
+publishing {
+    repositories {
+        maven {
+            val releasesRepoUrl = URI("https://oss.sonatype.org/service/local/staging/deploy/maven2")
+            val snapshotsRepoUrl = URI("https://oss.sonatype.org/content/repositories/snapshots/")
+            url = if (project.findProperty("release") == "true") releasesRepoUrl else snapshotsRepoUrl
+            credentials {
+                username = findProperty("mavenCentralUsername") as String?
+                password = findProperty("mavenCentralPassword") as String?
+            }
+        }
+    }
+    publications {
+        register<MavenPublication>("maven") {
+            groupId = "com.splunk"
+            artifactId = base.archivesName.get()
+
+            pom {
+                name.set("Splunk Otel Android")
+                description.set("A library for instrumenting Android applications for Splunk RUM")
```

Comment:
Shouldn't we set these in the libraries' `build.gradle.kts` files? I think the volley lib could have a bit different description

Reply:
Done, although I wonder if I approached it correctly with setting the name through an `extra`. WDYT?

Reply:
I think it's fine; the upstream repo handles a similar use case in a similar way: https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/conventions/src/main/kotlin/otel.javaagent-instrumentation.gradle.kts#L8